File: server/makc2022--nodejs--nestjs/src/layer3/nosql-mongo/sample/entities/dummy-tree-link/dummy-tree-link-entity.loader.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { nameof } from 'src/layer1/common/common-functions';
import { EntityLoader } from 'src/layer1/entity/entity-loader';
import { DummyTreeLinkEntityObject } from './dummy-tree-link-entity.object';

export class DummyTreeLinkEntityLoader extends EntityLoader<DummyTreeLinkEntityObject> {
  /** @inheritdoc */
  constructor(entityObject?: DummyTreeLinkEntityObject) {
    super(entityObject ?? ({} as DummyTreeLinkEntityObject));
  }

  /** @inheritdoc */
  override load(
    entityObject: DummyTreeLinkEntityObject,
    loadableProperties?: Set<string>
  ): Set<string> {
    const result = super.load(entityObject, loadableProperties);

    if (result.has(nameof<DummyTreeLinkEntityObject>('id'))) {
      this.entityObject.id = entityObject.id;
    }

    if (result.has(nameof<DummyTreeLinkEntityObject>('parentId'))) {
      this.entityObject.parentId = entityObject.parentId;
    }

    return result;
  }

  /** @inheritdoc */
  protected override createAllPropertiesToLoad(): Set<string> {
    return new Set<string>([
      nameof<DummyTreeLinkEntityObject>('id'),
      nameof<DummyTreeLinkEntityObject>('parentId'),
    ]);
  }
}

File: server/makc2022--dotnet/Makc2022.Layer3.Sql.Sample/Entities/DummyTreeLink/DummyTreeLinkEntityLoader.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2022.Layer3.Sql.Sample.Entity;

namespace Makc2022.Layer3.Sql.Sample.Entities.DummyTreeLink
{
    /// <summary>
    /// Загрузчик сущности "DummyTreeLink".
    /// </summary>
    public class DummyTreeLinkEntityLoader : EntityLoader<DummyTreeLinkEntityObject>
    {
        #region Constructors

        /// <inheritdoc/>
        public DummyTreeLinkEntityLoader(DummyTreeLinkEntityObject? entityObject = null)
            : base(entityObject ?? new DummyTreeLinkEntityObject())
        {
        }

        #endregion Constructors

        #region Public methods

        /// <inheritdoc/>
        public sealed override HashSet<string> Load(
            DummyTreeLinkEntityObject entityObject,
            HashSet<string>? loadableProperties = null)
        {
            var result = base.Load(entityObject, loadableProperties);

            if (result.Contains(nameof(EntityObject.Id)))
            {
                EntityObject.Id = entityObject.Id;
            }

            if (result.Contains(nameof(EntityObject.ParentId)))
            {
                EntityObject.ParentId = entityObject.ParentId;
            }

            return result;
        }

        #endregion Public methods

        #region Protected methods

        /// <inheritdoc/>
        protected override HashSet<string> CreateAllPropertiesToLoad()
        {
            return new HashSet<string>
            {
                nameof(EntityObject.Id),
                nameof(EntityObject.ParentId)
            };
        }

        #endregion Protected methods
    }
}
